### PR TITLE
fix #14314 do not analyze importc procs for effects

### LINKED ISSUE #14314: Getting "Warning: Cannot prove that 'result' is initialized" for an importcpp'd proc with `var T` return type
I need to add a trailing `=` to proc declarations for the functions I importcpp from std::vector, if I want to use `runnableExamples`. But if I do so, I get this warning if the return type is `var T`.

> Warning: Cannot prove that 'result' is initialized. This will become a compile time error in the future. [ProveInit]

### Example

Save below file to a `t.nim` and run `nim doc --backend:cpp t.nim` (requires the latest nim devel version).

```nim
{.push header: "<vector>".}

type
  Vector*[T] {.importcpp: "std::vector".} = object

proc newVector*[T](): Vector[T] {.importcpp: "std::vector<'*0>()", constructor.}

proc add*[T](v: var Vector[T], elem: T){.importcpp: "#.push_back(#)".}

proc first*[T](v: Vector[T]): var T {.importcpp: "front".} =
  ## Return the reference to the first element of the Vector.
  ##
  ## This has an alias proc `front`.
  ##
  ## https://en.cppreference.com/w/cpp/container/vector/front
  runnableExamples:
    var
      v = newVector[int]()
    v.add(100)
    doAssert v.first() == 100

{.pop.}
```

### Current Output

The doc gets generated fine, but we get this warning:

> /home/kmodi/sandbox/nim/bug_reports/cannot_prove_result_uninit/t.nim(11, 3) Warning: Cannot prove that 'result' is initialized. This will become a compile time error in the future. [ProveInit]

### Expected Output

No ProveInit warning

### Additional Information

* https://github.com/kaushalmodi/std_vector/issues/1#issuecomment-627085135

```
$ nim -v
Nim Compiler Version 1.3.3 [Linux: amd64]
Compiled at 2020-05-11
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 86669ef5b4c9711d81e961c2f68a05aa4f4d9ece
active boot switches: -d:release
```

/cc @timotheecour

### PR DESCRIPTION
**Pull request by timotheecour**
fix #14314

### CODE REVIEW

File: tests/trunner.nim [hunk 1]
```diff
@@ -122,3 +122,9 @@ else: # don't run twice the same test
     block: # mak sure --backend works with `nim r`
       let cmd = fmt"{nim} r --backend:{mode} --hints:off --nimcache:{nimcache} {file}"
       check execCmdEx(cmd) == ("ok3\n", 0)
+
+  block: # some importc tests
+    # issue #14314
+    let file = testsDir / "misc/mimportc.nim"
+    let cmd = fmt"{nim} r -b:cpp --hints:off --nimcache:{nimcache} --warningAsError:ProveInit {file}"
+    check execCmdEx(cmd) == ("witness\n", 0)
```
Comment (Clyybber) [inline tests/trunner.nim:130]: This doesn't need an extra "driver", afaict?
You can simply do it like other tests, with nimout and output or cmd

Comment (timotheecour) [reply tests/trunner.nim:130]: as is yes I agree, but I intended to grow this test upcoming PR's with things that are awkward to test via the testament spec DSL

**Comment by kaushalmodi** [comment]
@timotheecour This fix works beautifully! The std_vector tests seem to be sped up even more after this, may be because now it is not throwing the ProveInit warning for each importcpp proc.

**Comment by timotheecour** [comment]
good to hear!
> The std_vector tests seem to be sped up

I did not expect that, can you double check with some benchmark? If merely issuing warnings has a measurable performance impact on compile times, we have a bigger problem lol

**Comment by kaushalmodi** [comment]
As I said "The std_vector tests seem to be sped up". I measured and there's not much noticeable difference. May be lesser noise on the stdout is giving a feeling that the tests ended sooner. Based on the CPU load, it takes 6-8 seconds to finish `nim test` on the std_vector repo.

**Pull request merged**